lib/metadataTypes/MetadataType.js

```diff
@@ -534,16 +534,21 @@ class MetadataType {
      * @param {string} [overrideType] can be used if the API type differs from the otherwise used type identifier
      * @returns {Promise<{metadata:Util.MetadataTypeMap,type:string}>} Promise of item map
      */
-    static async retrieveSOAPgeneric(
+    static async retrieveSOAP(
         retrieveDir,
         buObject,
         requestParams,
         additionalFields,
         overrideType
     ) {
+        requestParams = requestParams || {};
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters
default potential?

Reply:
nope, because any given value would override it. we could define a single "thats ok too" value but what if null, undefined or empty string is passed in? all those eval to false and hence would trigger going into the alternative empty object